internal/handlers/handler_user_password.go

```diff
@@ -0,0 +1,51 @@
+package handlers
+
+import (
+	"fmt"
+
+	"github.com/authelia/authelia/internal/middlewares"
+	"github.com/authelia/authelia/internal/utils"
+)
+
+type updatePasswordRequestBody struct {
+	OldPassword string `json:"old_password"`
+	Password    string `json:"password"`
+}
+
+// UserPasswordPost handler for resetting passwords.
+func UserPasswordPost(ctx *middlewares.AutheliaCtx) {
+	userSession := ctx.GetSession()
+
+	var requestBody updatePasswordRequestBody
+	err := ctx.ParseBody(&requestBody)
+
+	if err != nil {
+		ctx.Error(err, unableToUpdatePasswordMessage)
+		return
+	}
+
+	ok, err := ctx.Providers.UserProvider.CheckUserPassword(userSession.Username, requestBody.OldPassword)
+	if !ok || err != nil {
+		ctx.Error(err, unableToUpdatePasswordMessage)
+		return
+	}
+
+	err = ctx.Providers.UserProvider.UpdatePassword(userSession.Username, requestBody.Password)
```

Comment:
I would prefer to do it this way, though it would require extending the interface, and coding both providers to implement a method which does it. If you'd like some help then let me know. The advantage is we can make this change as the actual LDAP user in the future doing it this way I believe. 

```suggestion
	err = ctx.Providers.UserProvider.ChangePassword(userSession.Username, requestBody.OldPassword, requestBody.Password)
```

For reference `ldap.NewPasswordModifyRequest` which is from the library we use takes this exact syntax.